packages/compass-schema/src/stores/reducer.ts

```diff
@@ -212,6 +214,12 @@ export const stopAnalysis = (): SchemaThunkAction<void> => {
   };
 };
 
+function isReadPreferenceSet(connectionString: string): boolean {
+  return !!new ConnectionString(connectionString).searchParams.get(
+    'readPreference'
+  );
+}
```

Comment:
Not a strong preference, but I feel like dataService might be a better place for this logic, it feels out of the domain of the schema plugin to know how to extract connection options from the connection info string.

Maybe we can make this an interface on the sample method? Something like `fallbackReadPreference` or `defaultReadPreference` option that we can pass to the method and will be only applied by the sample method using the same logic you implemented here? After all dataService is there to abstract away some parts of interacting with the driver for feature code